wlauto/workloads/geekbench/__init__.py

```diff
@@ -109,9 +109,13 @@ def __init__(self, device, **kwargs):
 
     def update_result(self, context):
         super(Geekbench, self).update_result(context)
-        major_version = versiontuple(self.version)[0]
-        update_method = getattr(self, 'update_result_{}'.format(major_version))
-        update_method(context)
+        # It is only possible to pull files from /data/data on rooted devices.
+        # Skip this step on devices without priveleges (e.g. Chromebooks) as the logs can be
+        # retrieved manually afterwards
+        if self.device.is_rooted:
```

Comment:
This will result in a "silent" failure on unrooted devices, and the users has to somehow know to pull the files manually afterwards. This manual step makes this workload useless for a large number of applications.

Instead, the workload should check if the device is rooted inside `initialize()` and fail with a `WorkloadError` if not. If it is still desirable to be able to run this on unrooted devices without generating results, this should be done via a parameter that the user has to explicitly enable.